spec/market_spec.rb

```diff
@@ -0,0 +1,95 @@
+require './lib/item'
+require './lib/vendor'
+require './lib/market'
+
+RSpec.describe Market do
+    before(:each) do
+        @market = Market.new("South Pearl Street Farmers Market")
+        @vendor1 = Vendor.new("Rocky Mountain Fresh")
+        @vendor2 = Vendor.new("Ba-Nom-a-Nom")
+        @vendor3 = Vendor.new("Palisade Peach Shack")
+        @item1 = Item.new({name: 'Peach', price: "$0.75"})
+        @item2 = Item.new({name: 'Tomato', price: "$0.50"})
+        @item3 = Item.new({name: "Peach-Raspberry Nice Cream", price: "$5.30"})
+        @item4 = Item.new({name: "Banana Nice Cream", price: "$4.25"})
+
+        @vendor1.stock(@item1, 35)
+        @vendor1.stock(@item2, 7)
+
+        @vendor2.stock(@item4, 50)
+        @vendor2.stock(@item3, 25)
+
+        @vendor3.stock(@item1, 65)
+        @vendor3.stock(@item3, 10)
+
+    end
+
+    it 'exists and has attributes' do
+        expect(@market).to be_a(Market)
+        expect(@market.name).to eq("South Pearl Street Farmers Market")
+        expect(@market.vendors).to eq([])
+    end
+
+    it 'can have vendors' do
+        @market.add_vendor(@vendor1)
+        @market.add_vendor(@vendor2)
+        @market.add_vendor(@vendor3)
+        expect(@market.vendors).to eq([@vendor1, @vendor2, @vendor3])
+    end
+    
+    it 'can tell the vendors name' do
+        @market.add_vendor(@vendor1)
+        @market.add_vendor(@vendor2)
+        @market.add_vendor(@vendor3)
+        expect(@market.vendor_names).to eq(["Rocky Mountain Fresh", "Ba-Nom-a-Nom", "Palisade Peach Shack"])
+    end
+
+    it 'can find vendors who sell a specific item' do
+        @market.add_vendor(@vendor1)
+        @market.add_vendor(@vendor2)
+        @market.add_vendor(@vendor3)
+        expect(@market.vendors_that_sell(@item1)).to eq([@vendor1, @vendor3])
+        expect(@market.vendors_that_sell(@item4)).to eq([@vendor2])
+    end
+
+    it "calculates potential_revenue" do
+        expect(@vendor1.potential_revenue).to eq(29.75)
```

Comment:
I like that you are testing different vendors (scenarios) and testing how the outputs differ.